### PR TITLE
update the shielded trc-20 contract doc

### PR DESCRIPTION
**Pull request by Federico2014**

### CODE REVIEW

**Comment by andelf** [comment]
+ missing ovk in `burn`

**Comment by Federico2014** [comment]
> * missing ovk in `burn`

In previous version, ovk is not needed. Now, the ovk is added.

**Comment by Federico2014** [comment]
> Should describe how to `approve` before mint.
> Should describe how to get an `rcm`
> Should describe how to get an `alpha`
> Why there's an `ovk` in mint?

ovk is needed in mint. the descriptions about approve, rcm and alpha are added

File: docs/mechanism-algorithm/shielded-TRC20-contract.md [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+       "note": {
+          "value": 50,
+          "payment_address": "ztron15js0jkuxczt8caq5hp59rnh6rgf34sek7vqn9u6ljelxv4nuzz2x9qe3ffm2wzz6ck53yxyhxs6",
+          "rcm": "74baec30dfac8ed59968955ff245ae002009005194e5b824c35ab88c52e5170e"
+       }
+    },
+    "shielded_TRC20_contract_address": "41f3392eaa7d38749176e0671dbc6912f8ef956943"
+}
+```
+Note: 1. the shielded TRC-20 contract should be deployed befor calling the api; 2. the `from_amount` is the real value, the `value` in note is the scaled value by `scalingFactor` set in the shielded TRC-20 contract, namely `from_amount` = `value` * `scalingFactor`. In this example, the value of `scalingFactor` is 100; 3. the `rcm` is a random number with the length of 32 bytes.
```
Comment (andelf) [inline docs/mechanism-algorithm/shielded-TRC20-contract.md:216]: `rcm` is a random field element from librustzcash.
It must be generated by `getrcm` call.

File: docs/mechanism-algorithm/shielded-TRC20-contract.md [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+Method: Post
+
+Parameters:
+
+```json
+{  
+    "ask": "0f63eabdfe2bbfe08012f6bb2db024e6809c16e8ed055aa41a6095424f192005",
+    "nsk": "cd43d722fd4b6b01f19449ea826c3e935609648520fcc2a95c0026f0fa9ee404",
+    "ovk": "0ff58efd75e083fe4fd759c8701e1c8cb6961c4297a12b2c800bdb7b2bcab889",
+    shielded_spends: [
```
Comment (andelf) [inline docs/mechanism-algorithm/shielded-TRC20-contract.md:272]: malformed json

File: docs/mechanism-algorithm/shielded-TRC20-contract.md [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+
+```
+{
+   "ask": "0f63eabdfe2bbfe08012f6bb2db024e6809c16e8ed055aa41a6095424f192005",
+   "tx_hash": "7fb99e66df052f430d50e80a3f197c319a4e35184ed48a433d6219e025741337",
+   "alpha": "d0f0768664f5c807f9e70d05993f029c71da529afaa5c037b44c63d6e13bc907"
+}
+```
+
+Note: `alpha` is a random number with the length of 32 bytes.
```
Comment (andelf) [inline docs/mechanism-algorithm/shielded-TRC20-contract.md:399]: Is `getrcm` required?

**Comment by andelf** [comment]
+ Add a description: This feature is now enabled on Nile Testnet.

File: docs/mechanism-algorithm/shielded-TRC20-contract.md (hunk 1 above)
Comment (Federico2014) [reply docs/mechanism-algorithm/shielded-TRC20-contract.md:216]: Yes, it's more accurate to say `rcm`  is the random scalar field element. I notice  the problem, but it's not easy to describe it accurately.

File: docs/mechanism-algorithm/shielded-TRC20-contract.md (hunk 3 above)
Comment (Federico2014) [reply docs/mechanism-algorithm/shielded-TRC20-contract.md:399]: `alpha` and `rcm` are generated by the same method.

**Pull request merged**